### PR TITLE
feat(unblock): add unblock rate limits

### PR DESCRIPTION
**Pull request by seanmonstar**
cc https://github.com/mozilla/fxa-auth-server/pull/1471

### CODE REVIEW

File: lib/email_record.js [hunk 1]
```diff
@@ -11,15 +11,17 @@ module.exports = function (limits, now) {
...
+    this.ub = []
   }
 
   EmailRecord.parse = function (object) {
     var rec = new EmailRecord()
     object = object || {}
     rec.bk = object.bk       // timestamp when the account was banned
     rec.rl = object.rl       // timestamp when the account was rate-limited
-    rec.xs = object.xs || [] // timestamps when emails were sent
+    rec.xs = object.xs || rec.xs // timestamps when emails were sent
```
Comment (rfk) [inline lib/email_record.js]: Does this fix a bug, or just for consistency?

**Comment by rfk** [comment]
@seanmonstar as an unsolicited f?, this LGTM at a high-level - it's more ad-hoc coupling between auth-server and customs-server APIs, but we're well down that road already, so 👍

File: lib/email_record.js (hunk 1 above)
Comment (seanmonstar) [inline lib/email_record.js]: I just noticed that an Array is already initialized in the constructor, why create a new one. One might hope that V8 can initialize arrays for nothing, but :shrug:

Comment (rfk) [inline lib/email_record.js]: 👍 ; mostly just confirming it wasn't a bug :-)

File: lib/server.js [hunk 2]
```diff
@@ -118,15 +118,16 @@ module.exports = function createServer(config, log) {
...
               return {
                 block: true,
                 retryAfter: ipRecord.retryAfter()
               }
             }
 
 
-            var blockEmail = emailRecord.update(action)
+            var wantsUnblock = req.body.payload && req.body.payload.unblockCode
+            var blockEmail = emailRecord.update(action, !!wantsUnblock)
```
Comment (rfk) [inline lib/server.js]: I think it would be worth adding a remote test exercises this payload-checking logic

**Comment by seanmonstar** [comment]
@rfk added a remote test that `payload.unblockCode` affects the `EmailRecord` in memcached.

**Pull request merged**